spec/semantic_conventions.md

```diff
@@ -7,55 +7,65 @@ operations used by applications. These conventions are used to populate the `att
+| `reranker.top_k`                       | Integer                     | 3                                                                          | Top K parameter of the reranker                                  |
+| `retrieval.documents`                  | List of objects<sup>†</sup> | `[{"document.id": "1", "document.score": 0.9, "document.content": "..."}]` | List of retrieved documents                                      |
+| `session.id`                           | String                      | `"26bcd3d2-cad2-443d-a23c-625e47f3324a"`                                   | Unique identifier for a session                                  |
+| `tag.tags`                             | List of strings             | ["shopping", "travel"]                                                     | List of tags to give the span a category                         |
+| `tool.description`                     | String                      | `"An API to get weather data."`                                            | Description of the tool's purpose and functionality              |
+| `tool.name`                            | String                      | `"WeatherAPI"`                                                             | The name of the tool being utilized                              |
+| `tool.parameters`                      | JSON string                 | `"{ 'a': 'int' }"`                                                         | The parameters definition for invoking the tool                  |
+| `tool_call.function.arguments`         | JSON string                 | `"{'city': 'London'}"`                                                     | The arguments for the function being invoked by a tool call      |
+| `tool_call.function.name`              | String                      | `get_current_weather`                                                      | The name of the function being invoked by a tool call            |
+| `user.id`                              | String                      | `"9328ae73-7141-4f45-a044-8e06192aa465"`                                   | Unique identifier for a user                                     |
+
+<sup>†</sup> To get a list of objects exported as OpenTelemetry span attributes, flattening of the list is necessary as shown in the example below.
+
+```python
+messages = [{"message.role": "user", "message.content": "hello"}, {"message.role": "assistant", "message.content": "hi"}]
+
+for i, obj in enumerate(messages):
+    for key, value in obj:
+        span.set_attribute(f"input.messages.{i}.{key}", value)
+```
+
+If the objects are further nested, flattening should continue until the attribute values are either simple values, i.e. `bool, str, bytes, int, float` or simple lists, i.e. `List[bool], List[str], List[bytes], List[int], List[float]` 
```

Comment:
can you add a typescript example too. We have some vendors watching our language compatibility